### PR TITLE
DEV: allow to have duplicate topic titles if category is different

### PR DESCRIPTION
**Pull request by sbernhard**

### CODE REVIEW

**Comment by coding-horror** [comment]
Is there discussion about this on meta.discourse.org? Because the URL slugs
would still be nearly identical, which is a bad thing, unless the category
is in the slug

**Comment by sbernhard** [comment]
See https://meta.discourse.org/t/could-we-check-the-duplicated-topic-title-for-the-each-category-only/98790/6

Same is true if allow duplicate titles is completely turned on. So, the limitation on the different category would still be a improvement.

**Comment by eviltrout** [comment]
This breaks backwards compatibility because the boolean setting is now a string. Anyone who changed it will have a bool in the database. 

At the very least we'll need a migration to migrate the old setting to the new one.

We should also investigate if any plugins use the setting as that code will be broken too.

**Comment by sbernhard** [comment]
That's true. Of course a migration can be added or would you prefer to have a new setting to configure this like:
Allow duplicate titles in different categories

**Comment by eviltrout** [comment]
The secondary setting does seem much easier.

**Comment by sbernhard** [comment]
Yes, much easier but the naming of the setting and that a user knows how to switch which setting is more difficult. Any suggestion is highly appreciated 😀

**Comment by sbernhard** [comment]
Added a new setting. 

Looks like the tests are failing because of another reason. Can you verfiy @eviltrout ?

File: config/locales/server.en.yml [hunk 1]
```diff
@@ -1449,6 +1449,7 @@ en:
     category_search_priority_very_high_weight: "Weight applied to ranking for very high category search priority."
     allow_uncategorized_topics: "Allow topics to be created without a category. WARNING: If there are any uncategorized topics, you must recategorize them before turning this off."
     allow_duplicate_topic_titles: "Allow topics with identical, duplicate titles."
+    allow_duplicate_topic_titles_at_different_category: "Allow topics with identical, duplicate titles if the category is different. (precondition: allow_duplicate_topic_titles=false)."
```
Comment (eviltrout) [inline config/locales/server.en.yml:1452]:
```suggestion
    allow_duplicate_topic_titles_at_different_category: "Allow topics with identical, duplicate titles if the category is different. allow_duplicate_topic_titles must be false."
```

File: config/site_settings.yml [hunk 2]
```diff
@@ -716,6 +716,7 @@ posting:
     default: true
     refresh: true
   allow_duplicate_topic_titles: false
+  allow_duplicate_topic_titles_at_different_category: false
```
Comment (eviltrout) [inline config/site_settings.yml:719]:
```suggestion
  allow_duplicate_topic_titles_category: false
```
This is a bit long. How about ^

**Comment by sbernhard** [comment]
All tests are finally green @eviltrout

**Pull request merged**